### PR TITLE
File Handling New

### PR DESCRIPTION
**Pull request by tuansondinh**
pimad-core version 0.0.1-alpha7
It's capable of filehandling and deleting PEAs.

### CODE REVIEW

**Comment by tuansondinh** [comment]
- private files have been removed from git history
- changed to previous test files
- ready for merge

**Review by julorenz — CHANGES_REQUESTED** [review]

File: src/PEAPool/PEAPool.ts [hunk 1]
```diff
@@ -74,14 +79,36 @@ abstract class APEAPool implements PEAPool {
...
-            this.responseHandler.handleCallbackWithResponse(PiMAdResponseTypes.ERROR, '', {}, callback);
+            // find pea by id
+            const localPEA: PEA | undefined  = this.peas.find(pea => identifier === pea.getPiMAdIdentifier());
+            // find index of pea
+            const index = this.peas.indexOf(localPEA as PEA,0);
+            // check if pea exists
+            if (localPEA) {
+                // delete
+                this.peas.splice(index, 1);
+                //TODO: Check if splice was successful?
```
Comment (cheidelbach) [inline src/PEAPool/PEAPool.ts:97]: Fix/Solve TODO

File: src/Converter/Importer/Importer.ts [hunk 2]
```diff
@@ -168,17 +168,17 @@ export class MTPFreeze202001Importer extends AImporter {
      */
     private accessDataSource(instructions: InstructionsConvertFrom, callback: (response: PiMAdResponse) => void): void {
         let gate: Gate = new MockGateFactory().create();
-        switch (instructions.source.slice(-4)) {
-            case '.aml':
+        switch (instructions.source.split('.').pop()) {
```
[the remark this answers is not in the document]
Comment (tuansondinh) [reply src/Converter/Importer/Importer.ts:171]: yes, pop() returns the last array element

**Comment by tuansondinh** [comment]
@cheidelbach thank you for your help 👍  ,
@julorenz can you merge, please?

File: src/PEAPool/PEAPool.ts [hunk 3]
```diff
@@ -94,7 +94,6 @@ abstract class APEAPool implements PEAPool {
             if (localPEA) {
                 // delete
                 this.peas.splice(index, 1);
-                //TODO: Check if splice was successful?
```
Comment (julorenz) [inline src/PEAPool/PEAPool.ts]: Why is that check not necessary?

I would expect this.peas.find() to return undefined now.

Comment (tuansondinh) [reply src/PEAPool/PEAPool.ts]: I have added a check :) but I'm not sure when that case could appear

**Review by julorenz — APPROVED** [review]

**Pull request merged**